UPGRADE.md

```diff
@@ -63,6 +63,45 @@ import { jsonServerRestClient } from 'admin-on-rest';
 import jsonServerRestClient from 'ra-data-json-server';
 ```
 
+## Introduced `i18nProvider` for loading of locale messages
+
+Previously you would supply `react-admin` with a `messages` prop object to allow alternative locales, this should be replaced with an `i18nProvider`. The new `i18nProvider` allows you to load the messages asynchronously. @See [`i18nProvider`](./Translation.md#i18nProvider) 
+
+```jsx
+// before
+import { enMessages } from 'admin-on-rest';
+const messages = { 'en': enMessages };
+
+<Admin locale='en' messages={messages} .../>
+
+// after
+
+import React from 'react';
+import { Admin, Resource, resolveBrowserLocale, GET_DEFAULT_MESSAGES, GET_LOCALE_MESSAGES } from 'react-admin';
+import frenchMessages from 'ra-language-french';
+import englishMessages from 'ra-language-english';
+
+const messages = {
+  en: () => englishMessages,
+  fr: () => import('ra-language-french').then( bundle => bundle.default ),
+};
+const i18nProvider = (type, params) => {
```

Comment:
This is already an advanced usage scenario. I'd rather keep this example the most simple possible using the default i18nProvider to have the same behavior as before. Then I'll introduce what this new provider allows one to do in a separate paragraph (something like *Lazily loading a locale messages*)